zebrad/src/commands/connect.rs

```diff
@@ -98,32 +98,31 @@ impl ConnectCmd {
 
 type Error = Box<dyn std::error::Error + Send + Sync + 'static>;
 
-struct Connect<ZN, ZNF, ZS> {
+struct Connect<ZN, ZS>
+where
+    ZN: Service<zebra_network::Request>,
+{
     retry_peer_set: tower::retry::Retry<zebra_network::RetryErrors, ZN>,
     peer_set: ZN,
     state: ZS,
     tip: BlockHeaderHash,
-    block_requests: FuturesUnordered<ZNF>,
+    block_requests: FuturesUnordered<ZN::Future>,
     requested_block_heights: usize,
     downloaded_block_heights: BTreeSet<BlockHeight>,
 }
 
-impl<ZN, ZNF, ZS, ZSF> Connect<ZN, ZNF, ZS>
+impl<ZN, ZS> Connect<ZN, ZS>
 where
-    ZN: Service<
-            zebra_network::Request,
-            Response = zebra_network::Response,
-            Error = Error,
-            Future = ZNF,
-        > + Send
+    ZN: Service<zebra_network::Request, Response = zebra_network::Response, Error = Error>
+        + Send
         + Clone
         + 'static,
-    ZNF: Future<Output = Result<zebra_network::Response, Error>> + Send,
-    ZS: Service<zebra_state::Request, Response = zebra_state::Response, Error = Error, Future = ZSF>
+    ZN::Future: Future<Output = Result<zebra_network::Response, Error>> + Send,
```

Comment:
```suggestion
    ZN::Future: Send,
```
does this bound work? the `Future<...>` is implied by `Service` i think

Reply:
oo, fancy